plone/app/discussion/browser/moderation.py

```diff
@@ -130,6 +134,25 @@ def allowed_transitions(self, obj=None):
             ]
             return transitions
 
+    def get_author_name(self, comment):
+        """Format author name with suffix for anonymous users.
+
+        Returns the author name with a suffix (Guest) appended for anonymous
+        comments. The suffix is translated to the current user's language.
+        """
+        author_name = (
+            comment.author_name
+        )  # If this is an anonymous comment (no creator), add the suffix
+        if not comment.creator and author_name:
+            # Translate the suffix to the current language
+            translated_suffix = translate(
+                self.anonymous_user_suffix, context=self.request
+            )
+            if not author_name.endswith(translated_suffix):
+                author_name = author_name + " " + translated_suffix
+
+        return author_name
```

Comment:
Can you create one function that is used both here and in comments.py, instead of repeating the code? Repetition makes it likely that a future fix will only happen in one of the 2 copies.